barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -629,8 +644,42 @@ fr ContentAddressedCachedTreeStore<LeafValueType>::get_current_root(ReadTransact
 // It is assumed that when these operations are being executed that no other state accessing operations
 // are in progress, hence no data synchronisation is used.
 
+template <typename LeafValueType> void ContentAddressedCachedTreeStore<LeafValueType>::commit_genesis_state()
```

Comment:
Some refactoring here. I created a specific method for commiting the genesis state.